contribs/drt/src/main/java/org/matsim/contrib/drt/optimizer/depot/Depots.java

```diff
@@ -64,13 +64,15 @@ public static boolean isSwitchingFromStopToStay(DvrpVehicle vehicle) {
 	}
 
 	public static Link findStraightLineNearestDepot(DvrpVehicle vehicle, Set<Link> links) {
-		Link currentLink = ((DrtStayTask)vehicle.getSchedule().getCurrentTask()).getLink();
+		Link currentLink = ((DrtStayTask) vehicle.getSchedule().getCurrentTask()).getLink();
 		return links.contains(currentLink) ?
-				null /* already at a depot*/ :
-				links.stream()
-						.min(Comparator.comparing(
-								l -> DistanceUtils.calculateSquaredDistance(currentLink.getToNode().getCoord(),
-										l.getFromNode().getCoord())))
-						.get();
+			null /* already at a depot*/ :
+			links.stream().map(l -> new DepotCandidates(l, DistanceUtils.calculateSquaredDistance(currentLink.getToNode().getCoord(),
+					l.getFromNode().getCoord())))
+				.sorted(Comparator.comparing(DepotCandidates::distance)
+					.thenComparing(h -> h.link.getId()))
+				.findFirst().get().link();
```

Comment:
Use `min(...).get()` instead of `sorted(...).findFirst().get()`. There is no point in sorting the whole stream if we only want to get the minimum element.

Reply:
Done